### PR TITLE
Update annotations for pending services

### LINKED ISSUE #1232: If you add a bundle and then move the services and then deploy the services move back to their original position
If you add mediawiki-scalable and then move the services around and then deploy, the services move back to their original x and y position.

### PR DESCRIPTION
**Pull request by makyo**
Fixes #1232

### CODE REVIEW

**Comment by makyo** [comment]
# QA

* Deploy a bundle
* Move a service
* Commit changes - service should stay where it was placed
* Deploy a service
* Move it
* Commit changes - service should stay where it was placed

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4234/
Test FAILed.

**Comment by makyo** [comment]
Rerunning: http://ci.jujugui.org:8080/job/juju-gui/4236

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4235/
Test PASSed.

File: jujugui/static/gui/src/app/views/topology/topology.js [hunk 1]
```diff
@@ -283,10 +283,17 @@ YUI.add('juju-topology', function(Y) {
...
-      if (box.pending) { return; }
-      this.get('env').update_annotations(
-          box.id, 'service', {'gui-x': box.x, 'gui-y': box.y});
-      box.inDrag = views.DRAG_ENDING;
+      if (box.pending) {
+        // If the service is pending, bypass the environment and set the service
+        // annotations directly.
+        var service = this.get('db').services.getById(box.id);
+        service.get('annotations')['gui-x'] = box.x;
+        service.get('annotations')['gui-y'] = box.y;
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/topology.js]: This is relying on the 'pass-by-reference' nature of JS. I think a marginally better way would be (I'll leave it up to you if you want to change):

```JavaScript
var annotations = service.get('annotations');
annotations['gui-x'] = box.x;
annotations['gui-y'] = box.y;
service.set('annotations', annotations);
```

**Comment by hatched** [comment]
:+1: With a trivial (up to you)
:ok: This appears to have resolved the issue as I could find. Great fix thx!

File: jujugui/static/gui/src/app/views/topology/topology.js (hunk 1 above)
Comment (makyo) [inline jujugui/static/gui/src/app/views/topology/topology.js]: Oh, that makes sense!  Will update.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4238/
Test PASSed.

**Comment by huwshimi** [comment]
QA OK :+1: nice work @makyo

**Comment by makyo** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**